### DISCUSSION TITLE
App Framework: Does it initiate rolling restarts?

### DISCUSSION THREAD

**Discussion by markusspitzli** [splunk/splunk-operator#1066] [Q&A] [answered]
Hi. 

I read in the [App Framework documentation](https://github.com/splunk/splunk-operator/blob/main/docs/AppFramework.md) the following Note, which I'm confused about. This note applies to the indexer and searchhead cluster. 

"Note: The Splunk cluster peer restarts are triggered by the contents of the Splunk apps deployed, and are not initiated by the App Framework."

I understand that the App Framework deploys the apps and reload the configs, but if a rolling restart is necessary, the App Framework won't trigger it. 

If this is correct, I would like to raise an enhancement request so that the App Framework is configurable to automatically initiate a rolling restart under certain conditions, like time of day or even week days or weekends. 
When it comes to automation, I would really appreciate seeing that the configurations are deployed and a restart initiated, so that I can be assured that the configs are active. Otherwise, the admin has to either intervene manually by triggering a restart or by debugging why a certain configuration isn't active.

Thanks for clarification
Markus

**Comment by sgontla** [comment, marked as answer]
@markusspitzli , What it means is, that the Operator Pod doesn't trigger any explicit Splunk instance restarts from App Framework point of view. However, when a bundle push is triggered by the App Framework, the ClusterManager(In the case of Indexer Cluster), Deployer(In the case of SHC) may trigger the rolling restarts, which is outside of the App Framework control.

**Reply by markusspitzli** [reply]
@sgontla Thanks for the clarification. I'm glad to hear that a restart maybe triggered if necessary. From experience of our huge Splunk Cluster, It is necessary to schedule restarts due to possible impact on either indexing or searching behavior. Do you think such a scheduling feature could be implemented by the Splunk Operator?

**Reply by sgontla** [reply]
@markusspitzli , Thank you. Sure, can you open an enhancement request, with the details about the use case?

**Reply by markusspitzli** [reply]
Issue is created: https://github.com/splunk/splunk-operator/issues/1084
